catalyst-gateway/bin/src/db/index/queries/staked_ada/get_txo_by_stake_address.rs

```diff
@@ -69,12 +82,64 @@ impl GetTxoByStakeAddressQuery {
     /// Executes a get txo by stake address query.
     pub(crate) async fn execute(
         session: &CassandraSession, params: GetTxoByStakeAddressQueryParams,
-    ) -> anyhow::Result<TypedRowStream<GetTxoByStakeAddressQuery>> {
-        let iter = session
-            .execute_iter(PreparedSelectQuery::TxoByStakeAddress, params)
+    ) -> anyhow::Result<Vec<GetTxoByStakeAddressQuery>> {
+        if session.is_persistent() {
+            if let Some(rows) = ASSETS_CACHE.get(&params.stake_address).await {
+                return Ok(rows);
+            }
+        }
+
+        let rows: Vec<_> = session
```

Comment:
rows needs to be in an ARC. See: https://github.com/moka-rs/moka/tree/main?tab=readme-ov-file#avoiding-to-clone-the-value-at-get